src/main/java/org/snomed/aag/data/services/ProjectAcceptanceCriteriaService.java

```diff
@@ -161,6 +166,17 @@ public ProjectAcceptanceCriteria findEffectiveCriteriaWithMandatoryItems(String
 			criteria.addToSelectedTaskCriteria(criteriaItem);
         }
 
+        // Join criteria items with matching authoring flag(s) in Branch metadata
+        Set<String> enabledAuthorFlags = MetadataUtil.getEnabledAuthorFlags(getBranchOrThrow(branchPath));
+        if (!enabledAuthorFlags.isEmpty()) {
+            for (CriteriaItem criteriaItem : criteriaItemService.findAllByEnabledByFlagInAndAuthoringLevel(enabledAuthorFlags, AuthoringLevel.PROJECT)) {
+                criteria.addToSelectedProjectCriteria(criteriaItem);
+            }
+            for (CriteriaItem criteriaItem : criteriaItemService.findAllByEnabledByFlagInAndAuthoringLevel(enabledAuthorFlags, AuthoringLevel.TASK)) {
+                criteria.addToSelectedTaskCriteria(criteriaItem);
+            }
+        }
+
```

Comment:
As the ticket is concerned with "including items with a matching flag", I added the logic to the findEffectiveCriteriaWithMandatoryItems method as it felt most appropriate.